pkgs/applications/audio/reaper/updater.sh

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash -p curl common-updater-scripts
+
+set -euo pipefail
+
+reaper_ver=$(curl -Ls https://www.reaper.fm/download.php | grep -o 'Version [0-9]\.[0-9]*' | head -n1 | cut -d' ' -f2)
+
+function set_hash_for_arch() {
+  local arch=$1
+  local suffix=$2
```

Comment:
I can't think of a better way to do this. The first update run updates both the hash and the version, then the second run fails because the version is already updated. But we need both to update different architectures...
If there's a less hacky way to do it, I'm interested to hear.

Reply:
The way you described it is a bit strange.

This derivation looks like two-in-one, because the `src` depends on the architecture. This way, this expr needs two updates.
Unfortunately I don't understand so well these updater scripts. 

Nonetheless I will throw an idea: treat all of them as one source, as if you need to download two files to build one final software.

Reply:
I went through a few other packages and it seems to be an unsolved issue in general :(
It looks like libreoffice has the best non-hacky solution which essentially does something like mine: https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/office/libreoffice/darwin/default.nix#L70

I'll do the same and see if I can make something more generic/reusable in the future.